src/main/java/com/example/demo/service/impl/BookServiceImpl.java

```diff
@@ -35,6 +35,27 @@ public BookDto findBookById(Long id) {
     @Override
     public BookDto createBook(CreateBookRequestDto requestDto) {
         Book book = bookMapper.toModel(requestDto);
-        return bookMapper.toDto(bookRepository.createBook(book));
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public BookDto updateBook(Long id, CreateBookRequestDto requestDto) {
+        Book existingBook = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id " + id)
+        );
+
+        existingBook.setTitle(requestDto.getTitle());
+        existingBook.setAuthor(requestDto.getAuthor());
+        existingBook.setIsbn(requestDto.getIsbn());
+        existingBook.setPrice(requestDto.getPrice());
+        existingBook.setDescription(requestDto.getDescription());
+        existingBook.setCoverImage(requestDto.getCoverImage());
```

Comment:
Create method in mapper
```suggestion
        bookMapper.updateBookFromDto(requestDto, existingBook);
```